### PR TITLE
[PLAT-1100] Create a new version from the most recent when copying

### PR DESCRIPTION
**Pull request by erinspace**
## Purpose

This will allow the most recent version of the cloned file to have the correct region, while the fileversion from the original file keeps its region

## Changes

- Create a new fileversion instead of adding FKs to all the old fileversions

## QA Notes

This should fix Lauren's most recent reported bug, and expected shouldnow happen!
Upload a file to region A
Give the file at least 2 versions
Copy it to region B
Upload a new version to region A
Try to render the file from region A
Expected
File would render.

## Documentation
none

## Side Effects
none

## Ticket
https://openscience.atlassian.net/browse/PLAT-1100

### CODE REVIEW

File: website/files/utils.py [hunk 1]
```diff
@@ -16,12 +18,19 @@ def copy_files(src, target_node, parent=None, name=None):
...
 
     if src.is_file and src.versions.exists():
-        cloned.versions.add(*src.versions.all())
-        most_recent_fileversion = cloned.versions.select_related('region').order_by('-created').first()
+        fileversions = src.versions.select_related('region').order_by('-created')
+        most_recent_fileversion = fileversions.first()
         if most_recent_fileversion.region != target_node.osfstorage_region:
+            # add all original version except the most recent
+            cloned.versions.add(*fileversions[1:])
+            # setting the id to None and calling save generates a new object
```
Comment (erinspace) [inline website/files/utils.py]: The way this copies the old fileversion is a little odd - got it from [the django docs](https://docs.djangoproject.com/en/dev/ref/models/instances/#how-django-knows-to-update-vs-insert) referenced from [this stackoverflow answer](https://stackoverflow.com/questions/4733609/how-do-i-clone-a-django-model-instance-object-and-save-it-to-the-database) - basically by setting the id to `None` it creates a new object instead of updating the old one.

I'm not totally set on this method over just doing `FileVersion.objects.create()` with all the other fileversion's attributes, as that maybe seems clearer?

Comment (felliott) [reply website/files/utils.py]: I'm not familiar w/ Django models, but how is the `src.clone()` method on line 12 implemented?  Is that a DRM-provided method, or is that something we added?  If the latter, what approach did we use?

Comment (erinspace) [reply website/files/utils.py]: AHHHHH I just assumed that was `Node` specific, but it's something we have on `BaseModel`! Woo! Thatttt does pretty much this but better. Thanks @felliott!!

PS: also fixing the failing test

File: website/files/utils.py [hunk 2]
```diff
@@ -16,12 +16,18 @@ def copy_files(src, target_node, parent=None, name=None):
...
     if src.is_file and src.versions.exists():
-        cloned.versions.add(*src.versions.all())
-        most_recent_fileversion = cloned.versions.select_related('region').order_by('-created').first()
-        if most_recent_fileversion.region != target_node.osfstorage_region:
-            most_recent_fileversion.region = target_node.osfstorage_region
-            most_recent_fileversion.save()
-
+        fileversions = src.versions.select_related('region').order_by('-created')
+        most_recent_fileversion = fileversions.first()
+        if most_recent_fileversion.region and (most_recent_fileversion.region != target_node.osfstorage_region):
```
Comment (erinspace) [inline website/files/utils.py]: Decided to add a check to make sure the fileversion has a region set before performing the other check because of the way we create and use FileVersions in our tests.  It seems like mostly a FileVersion is created from the factory first and then added to a file afterwards manually. When they're created since they're not attached to a `BaseFileNode` yet, there's not a good default region to set them to to match their file's nodesettings region.

So in short I decided to leave the failing test alone, and instead check for a region on the fileversion (tho they should always be there for real) so that future tests didn't have to worry about region on FileVersions when they don't need to.

File: website/files/utils.py [hunk 3]
```diff
@@ -16,12 +16,18 @@ def copy_files(src, target_node, parent=None, name=None):
...
     if src.is_file and src.versions.exists():
-        cloned.versions.add(*src.versions.all())
-        most_recent_fileversion = cloned.versions.select_related('region').order_by('-created').first()
-        if most_recent_fileversion.region != target_node.osfstorage_region:
-            most_recent_fileversion.region = target_node.osfstorage_region
-            most_recent_fileversion.save()
-
+        fileversions = src.versions.select_related('region').order_by('-created')
+        most_recent_fileversion = fileversions.first()
+        if most_recent_fileversion.region and most_recent_fileversion.region != target_node.osfstorage_region:
```
Comment (erinspace) [inline website/files/utils.py]: Decided to add a check to make sure the fileversion has a region set before performing the other check because of the way we create and use FileVersions in our tests. It seems like mostly a FileVersion is created from the factory first and then added to a file afterwards manually. When they're created since they're not attached to a BaseFileNode yet, there's not a good default region to set them to to match their file's nodesettings region.

So in short I decided to leave the failing test alone, and instead check for a region on the fileversion (tho they should always be there for real) so that future tests didn't have to worry about region on FileVersions when they don't need to.
